src/main/java/gdx/liftoff/Main.java

```diff
@@ -0,0 +1,582 @@
+        actor.addListener(handListener);
+    }
+
+    public static void addIbeamListener(Actor actor) {
+        actor.addListener(ibeamListener);
+    }
+
+    public static void addScrollFocusListener(Actor actor) {
+        actor.addListener(scrollFocusListener);
+    }
+
+    public static void onChange(Actor actor, Runnable runnable) {
+        actor.addListener(new ChangeListener() {
+            @Override
+            public void changed(ChangeEvent event, Actor actor) {
+                runnable.run();
+            }
+        });
+    }
+
+    public static void onClick(Actor actor, Runnable runnable) {
+        actor.addListener(new ClickListener() {
+            @Override
+            public void clicked(InputEvent event, float x, float y) {
+                runnable.run();
+            }
+        });
+    }
+
+    public static PopTable addTooltip(Actor actor, int align, String text) {
+        return addTooltip(actor, null, align, 0, text);
+    }
+
+    public static PopTable addTooltip(Actor actor, int align, float wrapWidth, String text) {
+        return addTooltip(actor, null, align, wrapWidth,  text);
+    }
+
+    public static PopTable addTooltip(Actor actor, Actor attachedActor, int align, float wrapWidth, String text) {
+        String style = (align & Align.bottom) != 0 ? "tooltip-arrow-up" : (align & Align.top) != 0  ? "tooltip-arrow-down" : (align & Align.left) != 0  ? "tooltip-arrow-right" : "tooltip-arrow-left";
+        PopTableTextHoverListener listener = new PopTableTextHoverListener(text, wrapWidth, align, align, skin, style);
+        listener.attachedActor = attachedActor;
+        actor.addListener(listener);
+        return listener.getPopTable();
+    }
+
+    public static PopTable addPopTableClickListener(Actor actor, int align, String description) {
+        return addPopTableClickListener(actor, null, align, 0, description);
+    }
+
+    public static PopTable addPopTableClickListener(Actor actor, int align, float wrapWidth, String description) {
+        return addPopTableClickListener(actor, null, align, wrapWidth,  description);
+    }
+
+    /**
+     * Utility method to attach a PopTable to an associated actor which will appear upon clicking the actor
+     * @param actor The actor to be clicked
+     * @param attachedActor The actor that the position of the PopTable will be relative to. This can differ from the
+     *                      actor
+     * @param align The alignment of the PopTable
+     * @param wrapWidth Set to 0 to not enable wrapping of the Label
+     * @param text The text to be added inside the PopTable
+     * @return
+     */
+    public static PopTable addPopTableClickListener(Actor actor, Actor attachedActor, int align, float wrapWidth, String text) {
+        String style = (align & Align.bottom) != 0 ? "tooltip-arrow-up" : (align & Align.top) != 0  ? "tooltip-arrow-down" : (align & Align.left) != 0  ? "tooltip-arrow-right" : "tooltip-arrow-left";
```

Comment:
This ternary is long and blank `@return` docs (this isn't the only one) seem like a waste of space. Instead of improving it, `validateUserData()` could be made into a massive conditional return. You need to decide whether to use the full width of the monitor or not - don't um and ah about it. Be bold, be strong!

Reply:
I adjusted that line and added descriptions to all params and returns. I'm keeping validateUserData() as is for legibility.